src/ibex_device_generator/utils/git_utils.py

```diff
@@ -116,7 +116,7 @@ def commit_all(self, msg: str) -> None:
         else:
             raise NothingToCommitError(self)
 
-    def create_submodule(self, name: str, url: str, path: str) -> None:
+    def create_submodule(self, name: str, url: str, path: str) -> None:  # pyright: ignore
```

Comment:
the overloaded class returns a "Submodule" object for that method which it can't replace with None. I couldn't see an easy way of creating a fake Submodule to be returned to nothing.

Reply:
Oh I see... That's not ideal overriding someone elses' class in an incompatible way like that...

I think that is pyright pointing out something we genuinely shouldn't be doing, but I guess 🤷 if you want to leave it (if you do want to fix it, I think just renaming the method where we use it to something that doesn't conflict with the base class would be sufficient)